lib/Sema/TypeCheckConstraints.cpp

```diff
@@ -992,6 +996,9 @@ namespace {
         return KPE;
       }
 
+      if (auto *init = simplifyTypeConstructionWithLiteralArg(expr))
```

Comment:
This should be guarded by Swift version 5 or later to avoid changing behavior for people compiling for earlier versions.

Reply:
I thought that we don't really want to do that, but maybe I'm wrong...

Reply:
I think we may have discussed seeing whether this broke anything in the compatibility suite and then discussion.

I think it's best to err on the side of caution, though, and guard changes in behavior like this under the next version so that existing code continues to compile with Swift 3 and Swift 4 releases.